### ISSUE TITLE
Bug(typescript-estree): never infers singleRun as true for project service

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#9854]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

typescript-estree

### Repro Code

```TypeScript
n/a
```

### ESLint Config

```javascript
import tseslint from "typescript-eslint";

export default tseslint.config(tseslint.configs.base, {
	files: ["**/*.ts"],
	languageOptions: {
		parserOptions: {
			projectService: true,
			tsconfigRootDir: import.meta.dirname,
		},
	},
	rules: {
		"@typescript-eslint/no-floating-promises": "error",
	},
});
```

### tsconfig

```jsonc
n/a
```

### Expected Result

Internally, `inferSingleRun` should return `true` when we're in single-run mode. Executing this config with something like `npx eslint .` should be determined to be in single-run mode.

### Actual Result

`singleRun` is always inferred to be `false` right now, because of:

https://github.com/typescript-eslint/typescript-eslint/blob/382e87b73644c138b3eade360a88ff2fff6af9c8/packages/typescript-estree/src/parseSettings/inferSingleRun.ts#L25-L33

### Additional Info

This doesn't generally have any impact on runtime performance right now. The project service doesn't generally use any single-run optimizations in single-run mode. But:

* It's confusing internally to see `singleRun` set to `false`
* There'll likely eventually be some optimizations, such as the caching TS system host mentioned in https://github.com/microsoft/TypeScript/issues/59338#issuecomment-2239701363 

---

We should also investigate whether the `options?.extraFileExtensions?.length` check can be bypassed with `options.projectService` per #9504:

https://github.com/typescript-eslint/typescript-eslint/blob/382e87b73644c138b3eade360a88ff2fff6af9c8/packages/typescript-estree/src/parseSettings/inferSingleRun.ts#L18-L23

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/typescript-estree` | `8.2.0` |
